src/laser_geometry.cpp

```diff
@@ -43,10 +43,20 @@ namespace laser_geometry
 
     // Fill the ranges matrix
     for (unsigned int index = 0; index < scan_in.ranges.size(); index++)
+    {
+      if (preservative)
+      {
+        ranges (0, index) = (double) (scan_in.ranges[index] == std::numeric_limits<float>::infinity()
+                          ? scan_in.range_max : scan_in.ranges[index]);
+        ranges (1, index) = (double) (scan_in.ranges[index] == std::numeric_limits<float>::infinity()
```

Comment:
None finite values in a pointcloud do not make any sense and have no usable meaning. 

If you don't change +Inf to a finite value, you will lose the relative vector of the ray when you convert to a pointcloud since (Inf, Inf) does not give you an angle.